core/src/main/scala/kafka/server/FetchSession.scala

```diff
@@ -758,16 +814,25 @@ class FetchManager(private val time: Time,
           case Some(session) => session.synchronized {
             if (session.epoch != reqMetadata.epoch) {
               debug(s"Session error for ${reqMetadata.sessionId}: expected epoch " +
-                s"${session.epoch}, but got ${reqMetadata.epoch} instead.");
+                s"${session.epoch}, but got ${reqMetadata.epoch} instead.")
               new SessionErrorContext(Errors.INVALID_FETCH_SESSION_EPOCH, reqMetadata)
+            } else if (session.usesTopicIds && reqVersion < 13 || !session.usesTopicIds && reqVersion >= 13)  {
```

Comment:
I noticed following warning message from our cluster (building on trunk). 

```
WARN [ReplicaFetcher replicaId=0, leaderId=1, fetcherId=0] Error in response for fetch request (type=FetchRequest, replicaId=0, ....(kafka.server.ReplicaFetcherThread:72)
org.apache.kafka.common.errors.FetchSessionTopicIdException: The fetch session encountered inconsistent topic ID usage
```

According to this code, changing the version in session is disallowed (please correct me if I misunderstood). Should fetch thread keep version in session meta? Or change the log level to `DEBUG`?

Reply:
Changing the version is not allowed. I'm not sure I follow what you mean by keeping version. Current usesTopicIds is set based on the version when the session is first created and maintained throughout the session. 

I think we will see this error transiently, but please let me know if you continue to see this issue after I fix the bug below.

Reply:
Thanks for explanation. Will test it after you fix the issue!

Reply:
This one too: https://issues.apache.org/jira/browse/KAFKA-13102
I have an idea of how to fix this one as well and it should make a big difference based on the testing I've done so far.